modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py

```diff
@@ -0,0 +1,149 @@
+        if sagemaker_project_id:
+            Tags.of(self).add("sagemaker:project-id", sagemaker_project_id)
+        if sagemaker_project_name:
+            Tags.of(self).add("sagemaker:project-name", sagemaker_project_name)
+
+        # TODO Add back cross-region support as a separate S3 replica module?
+        # sagemaker requires model package and inference image uri to be in the same region as model and endpoint
+        sagemaker.CfnModel.ContainerDefinitionProperty(model_package_name=model_package_arn)
+
+        monitor_image_uri = image_uris.retrieve(framework="model-monitor", region=self.region)
+
+        model_monitor_policy = iam.ManagedPolicy(
+            self,
+            "Model Monitor Policy",
+            document=iam.PolicyDocument(
+                statements=[
+                    iam.PolicyStatement(
+                        actions=[
+                            "s3:PutObject",
+                            "s3:GetObject",
+                            "s3:ListBucket",
+                        ],
+                        effect=iam.Effect.ALLOW,
+                        resources=[
+                            model_bucket_arn,
+                            f"{model_bucket_arn}/*",
+                        ],
+                    ),
+                    iam.PolicyStatement(
+                        actions=[
+                            "kms:CreateGrant",
+                            "kms:Encrypt",
+                            "kms:ReEncrypt*",
+                            "kms:GenerateDataKey*",
+                            "kms:Decrypt",
+                            "kms:DescribeKey",
+                        ],
+                        effect=iam.Effect.ALLOW,
+                        resources=[
+                            f"arn:{self.partition}:kms:{self.region}:{self.account}:key/*",
+                        ],
+                    ),
+                ]
+            ),
+        )
+
+        NagSuppressions.add_resource_suppressions(
+            model_monitor_policy,
+            suppressions=[
+                {
+                    "id": "AwsSolutions-IAM5",
+                    # TODO
+                    "reason": (
+                        "The IAM policy for create-ssm-param-provider role is already restricted "
+                        "to invoke only create-ssm-param-function"
+                    ),
+                },
+            ],
+        )
+
+        model_monitor_role = iam.Role(
+            self,
+            "Model Monitor Role",
+            assumed_by=iam.ServicePrincipal("sagemaker.amazonaws.com"),
+            managed_policies=[
+                model_monitor_policy,
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSageMakerFullAccess"),
+            ],
+        )
+
+        NagSuppressions.add_resource_suppressions(
+            model_monitor_role,
+            suppressions=[
+                {
+                    "id": "AwsSolutions-IAM4",
+                    # TODO
```

Comment:
what's this TODO for? (same goes for line 97)

Reply:
I believe a combination of:

1. Is there a better way to "fix" the nags.
2. The error strings I think don't make sense, I'll update those.

Reply:
I ended up improving the TODO comments, but I suspect we could reduce the wildcards here.